components/binding-usage-controller/docs/proposals/usage-kind.md

```diff
@@ -180,9 +187,11 @@ spec:
 7. The Usage Kind Protection Controller handles the ServiceBindingUsage deletion and checks, if the UsageKind is used by any ServiceBindingUsage. Then, the Usage Kind Protection Controller removes the UsageKind finalizer.
 8. Kubernetes removes the UsageKind instance.
 
+Without ServiceBindingUsage finalizers the implementation is not clean. The controller, which manages UsageKind finalizers, is triggered by the main controller instead of k8s which is not good. The cache in the indexer is not up to date (need investigation) and processing is not going through the working queue. It can cause a race condition. 
```

Comment:
`Without ServiceBindingUsage finalizers, the implementation is not clean. The protection controller, which manages UsageKind finalizers, is not triggered by Kubernetes but by the main controller. What is more, for the unknown reason, the cache in the indexer is not up to date and the processing does not go through the working queue, which can cause a race condition.`